### ISSUE TITLE
Injecting into private field of a base class doesn't work

### ISSUE THREAD

**Issue by ruslan-voodoo** [hadashiA/VContainer#359]
If I have a private field with the `Inject` attribute in a base class, it won't be injected upon resolving.
```
public class DataManager { }

public class BaseData
{
    [Inject] private DataManager _dataManager;
}

public class LevelData : BaseData { }

...

var builder = new ContainerBuilder();
            
builder.Register<DataManager>(Lifetime.Scoped).AsSelf();
builder.Register<LevelData>(Lifetime.Scoped).AsSelf();

var container = builder.Build();

var levelData = container.Resolve<LevelData>();
```
At this point, `levelData._dataManager` is null. But if I change private to protected, the field will be injected. 
Is this the desired behavior?

**Comment by kawai125** [comment]
The behavior is due to the C# language specification.

[VContainer's field injection is done in the generated extension methods](https://vcontainer.hadashikick.jp/resolving/container-api), but C# extension methods cannot access private variables in the type they are extending.

[Constructor Injection](https://vcontainer.hadashikick.jp/resolving/constructor-injection) or [Method Injection](https://vcontainer.hadashikick.jp/resolving/method-injection) can be used for Injection to private variables.

**Comment by YegorStepanov** [comment]
Either things have changed or you are not right. You can easily inject to private. 

```C#
var builder = new ContainerBuilder();

builder.Register<Class1>(Lifetime.Scoped).AsSelf();
builder.Register<Class2>(Lifetime.Scoped).AsSelf();

var container = builder.Build();

Debug.Log(container.Resolve<Class1>().Class2 == null); //false

public class Class1
{
    [Inject] private Class2 class2;

    public Class2 Class2 => class2;
}

public class Class2 { }

```

It looks that VContainer does not bypass base classes.

**Comment by YegorStepanov** [comment]
@hadashiA can you open it? It's a bug

**Comment by CamAlta** [comment]
This issue also affects private methods in base classes. 

vContainer uses `GetRuntimeMethods()` to discover methods decorated with the `[Inject]` attribute. 

The problem is that private methods in base classes are not returned from GetRuntimeMethods. Only protected or public methods are. 

```cd
public class ClassA 
{
  // Visible to GetRuntimeMethods
  protected void Foo()   {  }

  // Not visible to GetRuntimeMethods
  private void Bar() {}
}

public class ClassB : ClassA 
{
}
```

If you try to inject `Foo()` on a type of `ClassB` it won't be able to find it. 

[See this example fiddle for proof.](https://dotnetfiddle.net/Widget/T31Rk1)

**Comment by hadashiA** [comment]
Fixed in #468 #505
